lib/apiservers/engine/backends/image_cache.go

```diff
@@ -15,43 +15,48 @@
 package vicbackends
 
 import (
+	"encoding/json"
 	"fmt"
+	"net/http"
 	"os"
 	"sync"
 
 	log "github.com/Sirupsen/logrus"
 
+	derr "github.com/docker/docker/errors"
+	"github.com/docker/engine-api/types"
 	"github.com/vmware/vic/lib/apiservers/portlayer/client/storage"
 	"github.com/vmware/vic/lib/apiservers/portlayer/models"
 	"github.com/vmware/vic/lib/guest"
+	"github.com/vmware/vic/lib/metadata"
 )
 
-// LayerCache is an in-memory cache of image layers. It is refreshed at startup
+// ImageCache is an in-memory cache of image layers. It is refreshed at startup
 // by a call to the portlayer. It is updated when new layers are pulled or
 // layers are deleted.
-type LayerCache struct {
+type ImageCache struct {
 	sync.Mutex
 
-	// cache maps layer ID to layer data
-	cache map[string]*models.Image
+	// cache maps image ID to image metadata
+	cache map[string]*types.Image
```

Comment:
One comment.  We chatted about it over Slack, but I'll add it here for posterity.  types.Image doesn't have all the data we need.  ImageConfig does, but not types.Image.